src/ir_York.cpp

```diff
@@ -0,0 +1,340 @@
+void IRYorkAc::setOnTimer(const uint16_t nr_of_mins) {
+  _.OnTimer = nr_of_mins / 10;
+}
+
+/// Set the Off Timer value of the A/C.
+/// @param[in] nr_of_mins The number of minutes the timer should be.
+/// @note The timer time only has a resolution of 10 mins.
+/// @note Setting the Off Timer active will cancel the Sleep timer/setting.
+void IRYorkAc::setOffTimer(const uint16_t nr_of_mins) {
+  _.OffTimer = nr_of_mins / 10;
+}
+
+
+/// Get the On Timer setting of the A/C.
+/// @return The Nr. of minutes the On Timer is set for.
+uint16_t IRYorkAc::getOnTimer(void) const {
+  return _.OnTimer * 10;
+}
+
+/// Get the Off Timer setting of the A/C.
+/// @return The Nr. of minutes the Off Timer is set for.
+/// @note Sleep & Off Timer share the same timer.
+uint16_t IRYorkAc::getOffTimer(void) const {
+  return _.OffTimer * 10;
+}
+
+/// Convert the current internal state into its stdAc::state_t equivalent.
+/// @param[in] prev Ptr to the previous state if required.
+/// @return The stdAc equivalent of the native settings.
+stdAc::state_t IRYorkAc::toCommon(const stdAc::state_t *prev) const {
+  // TODO: all copy/pasted
+  stdAc::state_t result{};
+  // Start with the previous state if given it.
+  if (prev != NULL) {
+    result = *prev;
+  } else {
+    // Set defaults for non-zero values that are not implicitly set for when
+    // there is no previous state.
+    // e.g. Any setting that toggles should probably go here.
+    result.power = false;
+  }
+  result.protocol = decode_type_t::YORK;
+  result.mode = toCommonMode(_.Mode);
+  result.celsius = true;
+  result.degrees = getTemp();
+  result.fanspeed = toCommonFanSpeed(_.Fan);
+  result.swingv = _.SwingV ? stdAc::swingv_t::kAuto : stdAc::swingv_t::kOff;
+  // TODO: clarify sleep vs offtimer
+  result.sleep = -1;
+  // Not supported.
+  result.model = -1;
+  result.turbo = false;
+  result.swingh = stdAc::swingh_t::kOff;
+  result.light = false;
+  result.filter = false;
+  result.econo = false;
+  result.quiet = false;
+  result.clean = false;
+  result.beep = false;
+  result.clock = -1;
+  return result;
+}
+
+/// Convert the current internal state into a human readable string.
+/// @return A human readable string.
+String IRYorkAc::toString(void) const {
+  String result = "";
+  result.reserve(70);  // Reserve some heap for the string to reduce fragging.
+  result += addBoolToString(_.Power, kPowerStr);
+
+  result += addModeToString(_.Mode, kYorkAuto, kYorkCool,
+                            kYorkHeat, kYorkDry, kYorkFan);
+  result += addFanToString(_.Fan, kYorkFanHigh, kYorkFanLow,
+                           kYorkFanAuto, kYorkFanAuto,
+                           kYorkFanMedium);
+  result += addTempToString(getTemp(), true);
+  result += addBoolToString(_.SwingV, kSwingVStr);
+  result += addLabeledString(minsToString(_.OnTimer), kOnTimerStr);
+  result += addLabeledString(minsToString(_.OffTimer), kOffTimerStr);
```

Comment:
Change `_.OnTimer` to `getOnTimer()`.
Similar for the Off timer too.  You want the result to be in minutes, not 10's of minutes.